include/aspect/particle/output/interface.h

```diff
@@ -86,10 +86,37 @@ namespace aspect
                                const double current_time) = 0;
 
           /**
+           * Returns the particle file name taking into account the file index as well as the mpi proccess.
+           */
+          virtual
+          const std::string
+          get_file_name ();
```

Comment:
HDF5Output class does override this function as

```
      template <int dim>
      std::string
      HDF5Output<dim>::get_file_name () const
      {
        return "particles-"
               + Interface<dim>::get_file_index ()
               + this->output_file_suffix;
      }
```
mainly because at the moment, when running with multiple MPI processes, a single output file is written as if grouped files is set to 1. This isn;t the case for txt and vtu based output.